### ISSUE TITLE
Complex input types not mapped correctly

### ISSUE THREAD

**Issue by kaqqao** [graphql-java/graphql-java-tools#33]
If I have an input type that refers to another input type, the inner type doesn't get mapped correctly.
Example:

```
input AuthData {
    email: Email!
    password: String!
}

input Email {
    email: String
}
```

And the corresponding POJO:

```java
public class AuthData {

    private Email email;
    private String password;

   //gettes & setters for both fields
}
```

`Email` is just a simple _mutable_ POJO as well.

Trying to map this results in strange logs and an error:

> WARN com.coxautodev.graphql.tools.SchemaClassScanner - Schema type was defined but can never be accessed, and can be safely deleted: Email
> [WARNING] unavailable
> graphql.GraphQLException: type Email not found in schema

It first claims `Email` is unused, then it breaks because it's not found :/

I created a minimal project that demonstrates this here: https://github.com/kaqqao/graphql-java-tools-issue

I tried adding `Email` explicitly via `dictionary` but nothing changed. Also tried creating an `AuthDataResolver#email` method, but still nothing. I can't figure out if I'm doing something wrong or if this is a bug. The documentation on mapping inputs doesn't really exist and the example project doesn't have complex inputs...

**Comment by apottere** [comment]
This is fixed but hasn't been released yet, I'll cut one now.

**Comment by apottere** [comment]
Relevant issue: #26

**Comment by apottere** [comment]
This was just released in `3.1.3`
